resource_check.go

```diff
@@ -336,19 +337,42 @@ func resourceCheckCreate(d *schema.ResourceData, client interface{}) error {
 	if err != nil {
 		return fmt.Errorf("translation error: %w", err)
 	}
+
 	gotCheck, err := client.(*checkly.Client).Create(check)
 	if err != nil {
 		checkJSON, _ := json.Marshal(check)
 		return fmt.Errorf("API error 1: %w, Check: %s", err, string(checkJSON))
 	}
 	d.SetId(gotCheck.ID)
-	return resourceCheckRead(d, client)
+	return readCheck(d, client, false)
 }
 
 func resourceCheckRead(d *schema.ResourceData, client interface{}) error {
+	return readCheck(d, client, true)
+}
+
+func readCheck(d *schema.ResourceData, client interface{}, shouldSyncRemote bool) error {
 	check, err := client.(*checkly.Client).Get(d.Id())
 	if err != nil {
-		return fmt.Errorf("API error 2: could not read check %s, Error: %w", d.Id(), err)
+		if shouldSyncRemote && strings.Contains(err.Error(), "404") {
+			// the resource was deleted remotely, try to recreate it
+			{
```

Comment:
What do we have the extra brackets? Do you want to turn this into another func for readability?

Reply:
it's just for readability

Reply:
Could you turn this into a function? I find that easier to read personally

Reply:
I don't think it makes sense to turn this into a function, putting comment and belonging code in extra brackets is a style for easily indicating comment scope

Reply:
We can go with this then but, in general, I'm not a big fan of code blocks with brackets to comment and abstract code. I find methods and functions more suitable for this

Reply:
I agree, but depending on the situation, this code block basically does a simple `if then set` & it seem to belong to the local function scope logically, maybe the brackets here were too much, normally they make sense when the comment cover more than one statement, but here it's just about a single if statement